app/src/main/kotlin/compiler/parser/analysis/GrammarAnalysis.kt

```diff
@@ -12,6 +18,89 @@ class GrammarAnalysis<S : Comparable<S>> {
     }
 
     fun computeFollow(grammar: AutomatonGrammar<S>, nullable: Set<S>, first: Map<S, Set<S>>): Map<S, Set<S>> {
-        return TODO()
+
+        val result: MutableMap<S, MutableSet<S>> = HashMap()
+
+        var wasResultUpdated = true
+        fun update(setToUpdate: MutableSet<S>, secondSet: Set<S>) {
```

Comment:
Maybe rename function to `updateResult` and remove `setToUpdate` argument, since we'll be always updating result after applying modification in 4th comment.

Reply:
I'm not sure, I also use it in line 48 to update statesFirst

Reply:
I see, maybe you can rename `update` to `updateResultOrSetOnWhichResultDepends` or something similar to emphasize the reason behind setting `wasResultUpdated` flag in this function